src/dotty/tools/dotc/core/Substituters.scala

```diff
@@ -181,8 +181,13 @@ trait Substituters { this: Context =>
 
   final def substThis(tp: Type, from: RefinedType, to: Type, theMap: SubstRefinedThisMap): Type =
     tp match {
-      case tp @ RefinedThis(rt) =>
-        if (rt eq from) to else tp
+      case tp @ RefinedThis(rt, level) =>
+        if (rt eq from) 
+          to match { // !!! TODO drop
```

Comment:
We are not done yet. It will be removed in a later commit. So for now, just ignore the !!! TODOs.